### DISCUSSION TITLE
Load lowdefy yaml from URL

### DISCUSSION THREAD

**Discussion by gjones-r7** [lowdefy/lowdefy#496] [Questions and Help]
I'm looking to use lowdefy to dynamically generate a UI.  Can I load the lowdefy yaml from a URL? And can lowdefy periodically refetch the yaml to change the UI if it has changed? 

Love this project. Thanks.

**Comment by SamTolmay** [comment]
Hi @gjones-r7 

I'm not sure of your exact use case, so if you could give some more details I'll be able to give a more specific answer.

The way Lowdefy works you don't completely update the UI, but write the configuration to respond to changes in the data. You can use requests to fetch data from a URL. Have a look at [`AxiosHttp`](https://docs.lowdefy.com/AxiosHttp). This data can then be used in the UI, for example as data in a chart, or data to populate a Nunjucks template for the content of a markdown or HTML block. You can also use [lists](https://docs.lowdefy.com/lists) to render a set of blocks for each item in an array of data.

Currently, we don't have a way to execute requests on a timer, but this is something we definitely want to implement in the future.

The Lowdefy configuration (the blocks themselves) are not dynamic. They need to be defined in the configuration files and are added to the server at build time. This means if you need to make a change to these, the app needs to be built again and redeployed. However, modern CI tools like Netlify and Serverless make this easy to do. You can also use block visibility to hide blocks depending on the data.

Can you provide us with your specific use case? Then we can help you with the implementation.
